scipy/stats/_distribution_infrastructure.py

```diff
@@ -2705,7 +2703,7 @@ def _icdf_complement_safe(self, x, **params):
         out = self._icdf_complement(x, **params)
         eps = np.finfo(self._dtype).eps
         tol = self.tol if not _isnull(self.tol) else np.sqrt(eps)
-        mask = tol < eps/x
+        mask = tol * x < np.spacing(1 - x)
```

Comment:
I can't fully justify why I think `1 - x` is clear here and not above, but I think it has to do with the difference in what's going on here (even though it looks the same).

Above, we are concerned about the error in the output CDF, which can't (on average) be better than the half the spacing of the CCDF. (Right?) We run into trouble when the CCDF is close to 1 and the output CDF is close to zero.

Here, we're fixing the opposite problem: the user is specifying `x`, so the limited precision when `x` is close to `1.0` is their problem. We're concerned with the case that `x` is small and precise and we lose that precision when we take `1 - x`. We can't really estimate the error in the output... we're sort of interpreting `tol` as the relative uncertainty in the user's input.